web/blueprints/finance/__init__.py

```diff
@@ -123,18 +129,92 @@ def actions(activity_id):
 @nav.navigate(u"Bankkontobewegungen importieren")
 def bank_accounts_import():
     form = BankAccountActivitiesImportForm()
+    form.account.choices = [ (acc.id, acc.name) for acc in BankAccount.q.all()]
+    transactions = [] # new transactions which would be imported
+    old_transactions = [] # transactions which are already imported
     if form.validate_on_submit():
+        # login with fints
+        bankaccount = BankAccount.q.get(form.account.data)
         try:
-            csv_data = StringIO(form.csv_file.data.read().decode('utf-8'))
-            finance.import_bank_account_activities_csv(csv_data, form.expected_balance.data)
+            fints = FinTS3PinTanClient(
+                bankaccount.routing_number,
+                form.user.data,
+                form.pin.data,
+                bankaccount.fints_endpoint
+            )
+
+            acc = None
+            for account in fints.get_sepa_accounts():
+                if account.iban == bankaccount.iban:
+                    acc = account
+                    break
+            if acc is None:
+                raise KeyError('BankAccount with IBAN {} not found.'.format(
+                    bankaccount.iban)
+                )
+            if bankaccount.last_updated_at is not None:
+                start_date = bankaccount.last_updated_at.date()
+            else:
+                start_date = date(2018,1,1)
```

Comment:
For fetching the statements of a bank account FinTS requires a starting date. The official FinTS-specification states that the date can be any legal ISO 8601 date (which includes 0001-01-01). But trying 0001-01-01 as the starting date with an endpoint of my local Sparkasse returns an error. The first working date is 0010-01-01.
Since I don't know how other banks are handling this date, I decided to use 2018-01-01 as the starting date if no import was done in the past. I think this is ok because all previous transactions should be imported through the importer-scripts.